packages/runtime/src/account.ts

```diff
@@ -184,6 +184,58 @@ export class StoreAccount implements StoreAccountI {
     return asset.definitions;
   }
 
+  /**
+   * Modifies Asset fields
+   * @param assetId Asset Index
+   * @param fields Fields for modification
+   */
+  modifyAsset (assetId: number, fields: AssetModFields): void {
+    const asset = this.getAssetDef(assetId);
+    if (asset === undefined) {
+      throw new TealError(ERRORS.ASA.ASSET_NOT_FOUND, { assetId: assetId });
+    }
+    // check for blank fields
+    if ((fields.reserve !== "" && asset.reserve === "") || (fields.freeze !== "" && asset.freeze === "") ||
+      (fields.clawback !== "" && asset.clawback === "")) {
+      throw new Error("Cannot reset a blank address");
+    }
+
+    asset.manager = fields.manager;
+    asset.reserve = fields.reserve;
+    asset.freeze = fields.freeze;
+    asset.clawback = fields.clawback;
+  }
+
+  /**
+   * Freeze asset
+   * @param assetId Asset Index
+   */
+  freezeAsset (assetId: number, state: boolean): void {
+    const holding = this.assets.get(assetId);
+    if (holding === undefined) {
+      throw new TealError(ERRORS.ASA.ASSET_HOLDING_NOT_FOUND,
+        { address: this.address, assetId: assetId });
+    }
+    holding["is-frozen"] = state;
+  }
+
+  /**
+   * Destroys asset
+   * @param assetId Asset Index
+   */
+  destroyAsset (assetId: number): void {
+    const holding = this.assets.get(assetId);
+    const asset = this.getAssetDef(assetId);
+    if (holding === undefined || asset === undefined) {
```

Comment:
Isn't it possible that `holding` can be `undefined` ? What if account A created the asset, and accounts B, C opt-in. Then i do `destroyAsset`, in this case assetHolding of creator is undefined but i should be able to delete the asset, right?

Reply:
will check more, couldn't find about this

Reply:
yes, because algorand states that all the assets must be present in creator's account and that is only possible when creator opts-in. we can't destroy asset until it is fully present in asset creator's account